src/AzureMcp.csproj

```diff
@@ -22,6 +22,11 @@
     <NoWarn>$(NoWarn);SYSLIB0020</NoWarn>
   </PropertyGroup>
 
+  <!-- AOT compilation flags -->
+  <PropertyGroup>
+    <IlcFoldIdenticalMethodBodies>true</IlcFoldIdenticalMethodBodies>
```

Comment:
Because we are not publishing as an AOT native executable today, this flag will not affect the framework-dependent package we ship. This is a flag for AOT compiler when/if it is used.